src/Moq/MockException.cs

```diff
@@ -203,7 +203,35 @@ internal static MockException UnverifiedInvocations(IEnumerable<Invocation> invo
 					string.Join(Environment.NewLine, invocations)));
 		}
 
-		private MockExceptionReason reason;
+		/// <summary>
+		///   Returns the exception to be thrown when <see cref="Mock.VerifyNoOtherCalls(Mock)"/> finds invocations that have not been verified.
+		/// </summary>
+		internal static MockException UnverifiedInvocations(IEnumerable<MockException> errors)
+		{
+			Debug.Assert(errors.All(e => e.Reason == MockExceptionReason.UnverifiedInvocations));
+
+			return new MockException(
+				MockExceptionReason.UnverifiedInvocations,
+				string.Format(
+					CultureInfo.CurrentCulture,
+					Resources.UnverifiedInvocations,
+					string.Join(Environment.NewLine, errors.Select(error => error.Message))));
```

Comment:
There's two problems here:

1. Using `string.Format` on `Resources.UnverifiedInvocations` here will lead to duplicated text in the error message (since the same resource string will already be reproduced at the beginning of each `error.Message`.

   It would be better to follow the pattern that can be seen in the `MockException.UnmatchedSetups(IEnumerable<MockException> errors)` method:

   ```csharp
   return new MockException(
       MockExceptionReason.UnverifiedInvocations,
       string.Join(
           Environment.NewLine,
           errors.Select(error => error.Message)));
   ```

2. It would be important to include the failed mock's name in each original `error.Message` so that when multiple error messages are concatenated, the user can still tell which reported failures were caused by which mocks. This could be done as follows:

   * First, change the value of `Resources.UnverifiedInvocations` so it includes a placeholder for the mock's name: `The following invocations on mock {0} were not verified:\n{1}`.
   * In the `MockException.UnverifiedInvocations(IEnumerable<Invocation> invocations)` method, include the mock's name when `string.Format`-ting (which means that method needs an additional parameter for the mock:

      ```csharp
      string.Format(
          CultureInfo.CurrentCulture,
          Resources.UnverifiedInvocations,
          mock.ToString(),
          string.Join(Environment.NewLine, invocations)));
      ```